velox/vector/tests/MapVectorToRowVectorTest.cpp

```diff
@@ -0,0 +1,473 @@
+      .replaceNulls = false,
+      .throwOnDuplicateKeys = false,
+      .allowTopLevelNulls = false};
+  auto result = MapVector::toRowVector(*testCase, options, rows);
+
+  auto expectedType = ROW(outputFieldNames, {UNKNOWN(), UNKNOWN()});
+  ASSERT_EQ(*result->type(), *expectedType);
+  ASSERT_EQ(result->size(), testCase->size());
+
+  options.replaceNulls = true;
+  VELOX_ASSERT_THROW(
+      MapVector::toRowVector(*testCase, options, rows),
+      "Unsupported type for replacing nulls: UNKNOWN");
+}
+
+TEST_F(MapVectorToRowVectorTest, encoded) {
+  EXPECT_EQ(NonPOD::alive, 0);
+  auto testCases = makeTestCases();
+  auto reverseIndices = makeIndicesInReverse(testCases[0]->size());
+  auto reverseElementIndices =
+      makeIndicesInReverse(testCases[0]->mapKeys()->size());
+
+  // Wrap the top level map with a dictionary and a constant encoding.
+  for (auto& testCase : testCases) {
+    auto dictOnMap = BaseVector::wrapInDictionary(
+        nullptr, reverseIndices, testCase->size(), testCase);
+    executeTestCase(testCase);
+
+    auto constOnMap =
+        BaseVector::wrapInConstant(testCase->size(), 0, dictOnMap);
+    executeTestCase(testCase);
+  }
+
+  // Wrap the keys and values with a dictionary encoding.
+  for (auto& testCase : testCases) {
+    auto elementsSizes = testCase->mapValues()->size();
+    auto dictKeys = BaseVector::wrapInDictionary(
+        nullptr, reverseElementIndices, elementsSizes, testCase->mapKeys());
+    auto dictValues = BaseVector::wrapInDictionary(
+        nullptr, reverseElementIndices, elementsSizes, testCase->mapValues());
+    testCase->setKeysAndValues(dictKeys, dictValues);
+    executeTestCase(testCase);
+  }
+  testCases.clear();
+  EXPECT_EQ(NonPOD::alive, 0);
+}
+
+TEST_F(MapVectorToRowVectorTest, errorCases) {
+  SelectivityVector rows(1);
+  auto validMap = makeMapVectorFromJson<int64_t, int64_t>({"{1:1, 2:2, 3:3}"});
+
+  // Input of typeKind other than Map used
+  MapVector::ToRowVectorOptions options{
+      .keysToProject = {},
+      .outputFieldNames = {},
+      .replaceNulls = false,
+      .throwOnDuplicateKeys = false,
+      .allowTopLevelNulls = false};
+  ASSERT_ANY_THROW(
+      MapVector::toRowVector(*(validMap->mapKeys()), options, rows));
+
+  // Invalid key type
+  auto mapWithStringKeys =
+      makeMapVectorFromJson<std::string, int32_t>({"{\"x\": 99, \"y\": 100}"});
+  options.keysToProject = makeFlatVector<StringView>({"x", "y"});
+  options.outputFieldNames = {"x", "y"};
+  VELOX_ASSERT_THROW(
+      MapVector::toRowVector(*mapWithStringKeys, options, rows),
+      "Only SMALLINT, INTEGER, BIGINT keys are currently supported, instead got VARCHAR");
+
+  // Mismatched Type between keysToProject and map keys
+  VELOX_ASSERT_THROW(
+      MapVector::toRowVector(*validMap, options, rows),
+      "(BIGINT vs. VARCHAR) Key type and the type of keys to project are not the same");
+
+  // Keys to project are not empty
+  // Mismatched Type between keysToProject and map keys
```

Comment:
Comment doesn't seem to match code.

Reply:
copy-paste error, removed, thanks for catching this